src/tools/kdb/validate.cpp

```diff
@@ -0,0 +1,151 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include <validate.hpp>
+
+#include <cmdline.hpp>
+#include <kdb.hpp>
+#include <kdbio.hpp>
+#include <errors/errorFactory.hpp>
+#include <iostream>
+
+using namespace std;
+using namespace kdb;
+
+
+/**
+ * @brief Validate key database subtree
+ *
+ * Validate the part of the database that is rooted by the key given in the first argument.
+ * This is done by reading all key values, writing a different value, rewriting the original value and then re-setting the values
+ * in the key database. All loaded validation plugins are hereby triggered and their warnings are returned.
+ *
+ * @param cl the command line
+ *
+ * @return 0 if no warnings or errors were found and validation was therefore successful
+ * @return 1 if some warnings and errors occurred and validation therefore failed
+ */
+int ValidateCommand::execute (Cmdline const & cl)
+{
+	int argc = cl.arguments.size ();
+	if (argc != 1)
+	{
+		throw invalid_argument ("1 argument needed");
+	}
+
+	cout << "The given path was: " << cl.arguments[0] << endl;
+
+	KeySet ksUnfiltered;
+	/* use the given cmd line argument as the start key */
+	Key root = cl.createKey (0);
+
+	/* if -f (force) was given, the namespace is kept
+	 * and check-constraints in the spec:/ namespace
+	 * are not considered --> the key can be set to a
+	 * value that does not pass the validation criteria,
+	 * otherwise a cascading key is created for the
+	 * following kdb.get() */
+	// Key parentKey = cl.getParentKey (root);
+
+	// do not resume on any get errors
+	// otherwise the user might break
+	// the config
+	kdb.get (ksUnfiltered, root);
+
+	stringstream streamWarnings;
+	//printWarnings (streamWarnings, root, cl.verbose, cl.debug);
+
+	string strWarnings = streamWarnings.str ();
+	if (strWarnings.empty ()) //TODO: currently not working (always as with -f) because of disabled printWarnings above
+	{
+		cout << getFormattedSuccessString ("No warnings were issued! :)") << endl;
+	}
+	else
+	{
+		cerr << strWarnings;
+		root.clear ();
+
+		if (cl.force)
+		{
+			cout << getFormattedInfoString (
```

Comment:
again: only for -v.

Reply:
Fixed by commit 18edf392e8203989834a5c13debf5c1e4e7f8ab5.